sdk/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -368,6 +369,35 @@ abstract class ConverterMethods(stablePackages: language.StablePackages) {
     case _ => Left(s"Expected optional participant name but got $v")
   }
 
+  private def randomHex(rand: Random, length: Int) =
+    s"%0${length}x".format(rand.nextLong()).takeRight(length)
+
+  // If the givenIdHint is provided, we just use that as is.
+  def toPartyIdHint(
+      givenIdHint: String,
+      requestedDisplayName: String,
```

Comment:
Maybe consider Optional Strings, rather than checking for emptiness?

Reply:
Yes, I thought about that, but this is more faithful to the existing behaviour. Currently you can do 
```
allocatePartyWithHint "foo" (PartyIdHint "")
```
and it's treated equivalently to
```
allocateParty "foo"
```
This preserves that.